### PR TITLE
Making ingest_date column nullable on Submission model

### PR DESCRIPTION
**Pull request by sc15zs**
Ticket:
https://dluhcdigital.atlassian.net/browse/FPASF-588

Description:
As a first step of dropping the ingest_date column from the Submission, this PR makes the column nullable.

- [x] Unit tests and other appropriate tests added or updated
- [ ] README and other documentation has been updated / added (if needed)
- [x] Commit messages are meaningful and follow good commit message guidelines (e.g. "FS-XXXX: Add margin to nav items preventing overlapping of logo")

### CODE REVIEW

File: data_store/db/entities.py
Comment (samuelhwilliams) [inline data_store/db/entities.py:608]: Should we make this non-nullable at the same time? I've just checked prod and there are 0 records without a submission date - so this feels like a good time for us to start enforcing that?

There are a few tests that will fail because of this, but I think it should just be adding `submission_date` values for them.

**Comment by samuelhwilliams** [comment]
One comment, otherwise looks all good and I think should be fine to get in :) Nice start.

File: data_store/db/entities.py
Comment (sc15zs) [reply data_store/db/entities.py]: Changed submission_date to be non-nullable.

File: data_store/db/entities.py
Comment (samuelhwilliams) [inline data_store/db/entities.py:608]: Be careful - this is copying the same mistake that `ingest_date` was making.

I would suggest it's broadly OK to not set a default time, but if you do, it needs to be `datetime.now` - specifically _not_ called. I think it's safer to not have this default as the mistake is easily made.

**Review by samuelhwilliams — CHANGES_REQUESTED** [review]

**Review by samuelhwilliams — DISMISSED** [review]

File: tests/integration_tests/test_ingest_component_all_funds.py
Comment (samuelhwilliams) [inline tests/integration_tests/test_ingest_component_all_funds.py:141]: Please can you refer to Will's PR which puts a proper fix for this test in place :)

Comment (sc15zs) [reply tests/integration_tests/test_ingest_component_all_funds.py:141]: Does it mean to wait until his branch is merged in to main and then rebase my branch on main where this test is fixed, or to install freezegun and then Will to revert 90% of his PR?

Comment (samuelhwilliams) [reply tests/integration_tests/test_ingest_component_all_funds.py:141]: Will's PR is a better fix for the broken test than the implementation you've currently provided, IMO.

You can either review Will's PR and get it merged, or you could close his PR, cherry-pick the commit onto your branch and drop this slightly-hacky fix.

I think either is fine.

Comment (sc15zs) [reply tests/integration_tests/test_ingest_component_all_funds.py]: Done, thank you!

**Review by samuelhwilliams — APPROVED** [review]

**Pull request merged**
